### PR TITLE
Relax postmaster checks #51

### LINKED ISSUE #51: Relax the security warning for gmail
https://github.com/catalyst/moodle-tool_emailutils/blob/MOODLE_39_STABLE/classes/check/dnspostmastertools.php#L87

* [ ] Firstly if no emails go to gmail then this check doesn't matter
* [ ] This has some nuance because some domains are handled by gmail but are custom domain. Not sure its work doing the mx record lookup for these - cna be bonus points
* [ ] Even if all this is being used it should not be a warning, just info is ok
* [ ] However if it IS half setup then it should be a warning if its done wrong

### PR DESCRIPTION
**Pull request by bwalkerl**
Relaxed postmaster checks and added checking based on user email domains.

A couple of notes:
- The check on user domains is limited to the 5 most common for the mx lookup, but did we also want an additional check on gmail domains outside the top 5? If so should there be a minimum?
- Both tables in this check are rendered very roughly. Ideally they should use a proper template.

Closes #51

### CODE REVIEW

File: classes/dns_util.php [hunk 1]
```diff
@@ -96,6 +96,26 @@ public function get_subdomains($domain) {
...
+     * @param int $number limits the number of domains
+     * @return array
+     */
+    public function get_user_domains(int $number = 0): array {
+        global $DB;
+
+        $domainsql = $DB->sql_substr('LOWER(email)', $DB->sql_position("'@'", 'email') . ' + 1');
+        $sql = "SELECT $domainsql AS domain, count(*) AS count
+                  FROM {user}
+                 WHERE email LIKE '%@%.%' AND suspended = 0 AND deleted = 0
```
Comment (brendanheywood) [inline classes/dns_util.php]: just wondering about the LIKE and its performance if its needed. If we had a dodgy email in the system then in theory it never gets confirmed so maybe we can just filter on confirmed = 1 and that's all we need?

File: classes/dns_util.php [hunk 2]
```diff
@@ -96,6 +96,26 @@ public function get_subdomains($domain) {
...
+              GROUP BY $domainsql
+              ORDER BY count DESC";
+        if (!empty($number)) {
+            $sql .= " LIMIT $number";
```
Comment (brendanheywood) [inline classes/dns_util.php]: the limit should be a param to get_records_sql to be cross-db

File: classes/dns_util.php (hunk 1 above)
Comment (bwalkerl) [reply classes/dns_util.php]: In one of my test sites I do have a handful of accounts that are confirmed without valid emails, one of them being the guest user. The others are possibly related to some manual change, but removing the LIKE opens it up the possibility of some weird results.

Comment (brendanheywood) [reply classes/dns_util.php]: ok cool

**Pull request merged**